src/erd-script.sql

```diff
@@ -0,0 +1,255 @@
+);
+
+-- Product Images Table
+CREATE TABLE product_images (
+  id SERIAL PRIMARY KEY,
+  product_id INT NOT NULL,
+  image_url TEXT NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  FOREIGN KEY (product_id) REFERENCES products (id)
+);
+
+CREATE TYPE ORDER_STATUS AS ENUM ('PENDING', 'COMPLETED', 'CANCELLED');
+
+-- Shopping Cart Table
+CREATE TABLE shopping_cart (
+  id SERIAL PRIMARY KEY,
+  user_id INT NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  FOREIGN KEY (user_id) REFERENCES users (id)
+);
+
+-- Cart Items Table
+CREATE TABLE shopping_cart_items (
+  id SERIAL PRIMARY KEY,
+  cart_id INT NOT NULL,
+  product_id INT NOT NULL,
+  quantity INTEGER NOT NULL CHECK (quantity > 0),
+  price_at_time DECIMAL(10, 2) NOT NULL,
+  created_at TIMESTAMP(3) DEFAULT CURRENT_TIMESTAMP NOT NULL,
+  UNIQUE (cart_id, product_id),
+  FOREIGN KEY (cart_id) REFERENCES shopping_cart (id),
+  FOREIGN KEY (product_id) REFERENCES products (id)
+);
+
+-- Orders Table
+CREATE TABLE orders (
+  id SERIAL PRIMARY KEY,
+  client_id INT NOT NULL,
+  total_amount DECIMAL(10, 2) NOT NULL,
+  status ORDER_STATUS NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  FOREIGN KEY (client_id) REFERENCES users (id)
+);
+
+-- Order Items Table
+CREATE TABLE order_items (
+  id SERIAL PRIMARY KEY,
+  order_id INT NOT NULL,
+  product_id INT NOT NULL,
+  quantity INT NOT NULL,
+  unit_price DECIMAL(10, 2) NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  updated_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  FOREIGN KEY (order_id) REFERENCES orders (id),
+  FOREIGN KEY (product_id) REFERENCES products (id)
+);
+
+-- Liked Products Table
+CREATE TABLE liked_products (
+  id SERIAL PRIMARY KEY,
+  client_id INT NOT NULL,
+  product_id INT NOT NULL,
+  created_at TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  FOREIGN KEY (client_id) REFERENCES users (id),
+  FOREIGN KEY (product_id) REFERENCES products (id)
+);
+
+-- Enum for payment status
+CREATE TYPE PAYMENT_STATUS AS ENUM (
+  'PENDING',
+  'PAID',
+  'FAILED',
+  'REFUNDED',
+  'CANCELED'
+);
+
+-- Enum for payment method
+CREATE TYPE PAYMENT_METHOD AS ENUM (
+  'CREDIT_CARD',
+  'DEBIT_CARD',
+  'BANK_TRANSFER',
+  'PAYPAL',
+  'CASH',
+  'CRYPTO'
+);
+
+-- Stripe Payments table
+CREATE TABLE stripe_payments (
+  id SERIAL PRIMARY KEY,
+  order_id INT NOT NULL,
+  stripe_payment_intent_id VARCHAR(255) NOT NULL,
+  status PAYMENT_STATUS NOT NULL DEFAULT 'PENDING',
+  payment_method PAYMENT_METHOD NOT NULL,
+  payment_date TIMESTAMP DEFAULT CURRENT_TIMESTAMP,
+  FOREIGN KEY (order_id) REFERENCES orders (id)
+);
+
+-- Webhook log table to track Stripe events
+CREATE TABLE stripe_webhooks (
+  id SERIAL PRIMARY KEY,
+  event_id VARCHAR(255) UNIQUE NOT NULL,
+  event_type VARCHAR(255) NOT NULL,
+  payload JSONB,
```

Comment:
is there any sort of sensitive information that gets sent here? If so storing the plain JSON could be unsafe and it may need to be hashed. just something for you to think/review later